### DISCUSSION TITLE
Memory damage from too many write cycles

### DISCUSSION THREAD

**Discussion by Pechi1** [emsesp/EMS-ESP32#2282] [Q&A]
Hello,
I installed my new CS6800i heat pump three weeks ago and have been adjusting the maximum compressor power and other parameters in PV mode via EMS-ESP.

Does anyone know how these settings are stored in the Bosch control unit? They must be saved in some form of non-volatile memory, possibly an EEPROM. I'm concerned that frequently changing these values might eventually cause issues with the memory over time.

**Comment by proddy** [comment]
nope, sorry!

**Comment by moritzthecat** [comment]
this is a very valid question. I would expect that all control parameters being calculated are RAM, but config parameters are non-volatile and may not have endurance optimization in the firmwarre for frequent writes.
